src/sqlfluff/core/parser/segments/base.py

```diff
@@ -155,7 +155,11 @@ def __eq__(self, other):
 
     def __hash__(self):
         return hash(
-            (self.__class__.__name__, self.raw, self.pos_marker.source_position())
+            (
```

Comment:
Slightly unrelated fix, but this fixes an issue where segments lacking a `pos_marker` can't be hashed (thus can't be added to sets, etc.)